spec/features/variable_drafts/create_variable_draft_with_associated_collection_spec.rb

```diff
@@ -0,0 +1,176 @@
+      end
+
+      it 'associates the collection upon draft creation' do
+        within '.nav-top' do
+          click_on 'Done'
+        end
+        within '#invalid-draft-modal.eui-modal-content' do
+          click_on 'Yes'
+        end
+        within '.collection-association' do
+          click_on 'Collection Association'
+        end
+        within '#variable-draft-collection-association-table tbody' do
+          expect(page).to have_content('CIESIN_SEDAC_ANTHROMES_v2_1700')
+          expect(page).to have_css('tr', count: 1)
+        end
+      end
+    end
+
+    context 'when the supplied concept-id is not found' do
+      before do
+        fill_in 'associated_collection_id', with: 'C1-INVALID'
+        click_on 'Create New Record'
+      end
+
+      it 'does not navigate to the new_variable_draft_path' do
+        expect(page).to have_no_content('Variable Information')
+        expect(page).to have_no_css('.umm-form')
+        expect(page).to have_no_css('.variable-form')
+      end
+
+      it 'notifies the user that the concept supplied was not found' do
+        expect(page).to have_content("No matches were found for C1-INVALID")
+      end
+    end
+
+    context 'when the current provider context is not congruent with the supplied concept-id' do
+      before do
+        fill_in 'associated_collection_id', with: not_current_provider_concept_id
+        click_on 'Create New Record'
+      end
+
+      it 'does not navigate to the new_variable_draft_path' do
+        expect(page).to have_no_content('Variable Information')
+        expect(page).to have_no_css('.umm-form')
+        expect(page).to have_no_css('.variable-form')
+      end
+
+      it 'notifies the user that the provider context needs to be changed' do
+        expect(page).to have_content("Variables can only be associated to collections within the same provider. To create a variable for #{not_current_provider_concept_id} you must change your provider context.") 
+      end
+    end
+
+    context 'when the cmr search for supplied concept-id is unsuccessful' do
+      before do
+        error_body = '{"errors": ["Error searching CMR", "2nd message"]}'
+        error_response = Cmr::Response.new(Faraday::Response.new(status: 401, body: JSON.parse(error_body), response_headers: {}))
+        allow_any_instance_of(Cmr::CmrClient).to receive(:get_collections_by_post).and_return(error_response)
+        fill_in 'associated_collection_id', with: collection_concept_id
+        click_on 'Create New Record'
+      end
+
+      it 'surfaces the first CMR error message' do
+        expect(page).to have_content('Error searching CMR')
+      end
+
+      it 'does not navigate to the new_variable_draft_path' do
+        expect(page).to have_no_content('Variable Information')
+        expect(page).to have_no_css('.umm-form')
+        expect(page).to have_no_css('.variable-form')
+      end
+    end
+
+    context 'when no concept-id is supplied' do
```

Comment:
I thought that this would have been covered in the manage variable tests, but I didn't see one covering it.  This one does make me wonder if all of these tests should be in the manage variables files.

Reply:
The reason I put everything with the variable drafts files was because the spec in the manage_variables folder doesn't test the creation of variable drafts, it just makes sure the open drafts on the manage variables page is being rendered properly. Let me know what you think I should do.